src/Core.hs

```diff
@@ -107,6 +107,7 @@ data CoreF pat core
   | CoreIdentEq HowEq core core         -- bound-identifier=? :: Syntax -> Syntax -> Macro Bool
                                         -- free-identifier=?  :: Syntax -> Syntax -> Macro Bool
   | CoreLog core
+  | CoreWhichProblem
```

Comment:
Now that I experienced first hand how many places I need to edit in order to add a new macro action, I agree that the next step should be to make it easier to introduce new macro actions :)